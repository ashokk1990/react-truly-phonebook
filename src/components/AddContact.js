import PropTypes from 'prop-types';
import React from 'react';


class AddContact extends React.Component {
    handleAddSubmit = event => {
        event.preventDefault();
        for (const v of Object.values(this.state)) {
            if (!v) return;
        }

        this.props.handleSubmit(this.state);
    };
    handleOnChange = (name, value) => {
        this.setState({[name]: value});
    };

    constructor(props) {
        super(props);
        this.state = {
            name: "",
            number: "",
            context: "",
        }
    }

    render() {

        return (
            <div className="ui form">
                <div className="fields">
                    <div className="field">
                        <label>Name</label>
                        <input type="text" placeholder="Name"
                               onChange={(e) => this.handleOnChange("name", e.target.value)} required/>
                    </div>
                    <div className="field">
                        <label>Number</label>
                        <input type="number" placeholder="Number"
                               onChange={(e) => this.handleOnChange("number", e.target.value)} required/>
                    </div>
                    <div className="field">
                        <label>Context</label>
                        <input type="text" placeholder="Context"
                               onChange={(e) => this.handleOnChange("context", e.target.value)} required/>
                    </div>
                </div>
                <button className="ui icon button" onClick={this.handleAddSubmit}>
                    Add
                </button>
            </div>
        )
    }
}

AddContact.propTypes = {
    handleSubmit: PropTypes.func.isRequired
}

export default AddContact
