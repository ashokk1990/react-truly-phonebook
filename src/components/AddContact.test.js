import {shallow} from 'enzyme';
import React from "react";
import AddContact from "./AddContact";

const mockhandleSubmit = jest.fn();

const props = {
    handleSubmit: mockhandleSubmit,
};

describe('AddContact Component', () => {
    let addContactComponent = shallow(<AddContact {...props} />);
    it('snapshot', () => {
        expect(addContactComponent).toMatchSnapshot();
    });

    it('handleOnChange for correct text ', () => {
        addContactComponent.instance().handleOnChange("name", "Bob Barker");
        expect(addContactComponent.state().name).toEqual('Bob Barker');
    });

    it('handleOnChange for correct text ', () => {
        addContactComponent.instance().handleOnChange("number", "+12675558080");
        expect(addContactComponent.state().number).toEqual('+12675558080');
    });

    it('handleOnChange for correct text ', () => {
        addContactComponent.instance().handleOnChange("context", "personal");
        expect(addContactComponent.state().context).toEqual('personal');
    });

    it('should call handleSubmit method when button is clicked', () => {
        const event = {
            preventDefault: () => {
            }
        }
        addContactComponent.find('button').prop('onClick')(event);
        expect(mockhandleSubmit).toHaveBeenCalledWith({
            "name": "Bob Barker",
            "number": "+12675558080",
            "context": "personal"
        });
    });
});
