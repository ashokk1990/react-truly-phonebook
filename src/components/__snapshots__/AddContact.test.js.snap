// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddContact Component snapshot 1`] = `
<div
  className="ui form"
>
  <div
    className="fields"
  >
    <div
      className="field"
    >
      <label>
        Name
      </label>
      <input
        onChange={[Function]}
        placeholder="Name"
        required={true}
        type="text"
      />
    </div>
    <div
      className="field"
    >
      <label>
        Number
      </label>
      <input
        onChange={[Function]}
        placeholder="Number"
        required={true}
        type="number"
      />
    </div>
    <div
      className="field"
    >
      <label>
        Context
      </label>
      <input
        onChange={[Function]}
        placeholder="Context"
        required={true}
        type="text"
      />
    </div>
  </div>
  <button
    className="ui icon button"
    onClick={[Function]}
  >
    Add
  </button>
</div>
`;
